### PR TITLE
Job activity history plumb the data

### PR DESCRIPTION
**Pull request by amitaekbote**
Use jooq+flyway instead of only jooq
Move jooqflyway to JAH component, we can discuss as a team if this needs to be moved out later, I prefer it here till I work out the bugs that will come with it
Read from publisher queue/write to job activity store
Use local postgres for testing, I am not entirely sure why embedded postgres nopes out on some mac environments, I think setting up local postgres for testing this on everyones dev setups is a lot to do right now. I have a jira to move to test containers which should help solve this problem

### CODE REVIEW

File: titus-supplementary-component/job-activity-history/build.gradle [hunk 1]
```diff
@@ -1,8 +1,71 @@
...
+
+    compile 'com.opentable.components:otj-pg-embedded:0.13.+'
+    // Hikari DataSource for connection pooling to the DB
+    compile 'com.zaxxer:HikariCP:3.2.0'
     compile project(':titus-server-runtime')
 
     testCompile project(':titus-testkit')
 }
+
+// Begin: Jooq code generation using EmbeddedPostgres database and Flyway migration
```
Comment (tbak) [inline titus-supplementary-component/job-activity-history/build.gradle:47]: We keep storage implementation in separate subproject (in titus-ext/) so we can easily swap it if needed with different implementation.

File: titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/JobActivityWorker.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        this.clock = titusRuntime.getClock();
+        this.titusRuntime = titusRuntime;
+        this.jobActivityStore = jobActivityStore;
+
+        Registry registry = titusRuntime.getRegistry();
+    }
+
+    @Activator
+    public void enterActiveMode() {
+        ScheduleDescriptor scheduleDescriptor = ScheduleDescriptor.newBuilder()
```
Comment (tbak) [inline titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/JobActivityWorker.java:42]: There are more parameters to be configured, like interval, executor, etc.

File: titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/store/JooqConfigurationBean.java [hunk 3]
```diff
@@ -14,36 +14,34 @@
...
 
 import com.netflix.archaius.api.annotations.Configuration;
 import com.netflix.titus.common.util.SpringConfigurationUtil;
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.core.env.Environment;
 
-@Configuration(prefix = "titus.ext.jooqflyway")
-public class JooqConfiguration {
+@Configuration(prefix = "titus.ext.supplementary.jobactivity")
+public class JooqConfigurationBean implements JooqConfiguration {
```
Comment (tbak) [inline titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/store/JooqConfigurationBean.java:25]: This is not needed anymore. You can use `Archaius2Ext.newConfiguration` to create Archaius style proxy to Spring config.

File: titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/store/JooqJobActivityContextComponent.java [hunk 4]
```diff
@@ -44,67 +41,73 @@ public SQLDialect dialect() {
...
         hikariConfig.setAutoCommit(true);
+        System.err.println("PRODUCER");
 
         // Connection management
         hikariConfig.setConnectionTimeout(10000);
         hikariConfig.setMaximumPoolSize(10);
         hikariConfig.setLeakDetectionThreshold(3000);
 
         if (jooqConfiguration.isInMemoryDb()) {
+            System.out.println("IN memory");
```
Comment (tbak) [inline titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/store/JooqJobActivityContextComponent.java:87]: logger?

File: titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/store/JooqJobActivityStore.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    public Mono<JobActivityPublisherRecord> readRecordFromPublisherQueue() {
+        return JooqUtils.executeAsyncMono(() -> {
+            long startTimeMs = System.currentTimeMillis();
+            List<JobActivityPublisherRecord> records = producerDSLContext
+                    .selectFrom(ACTIVITY_QUEUE)
+                    .orderBy(ACTIVITY_QUEUE.QUEUE_INDEX.asc())
+                    .limit(1)
+                    .fetchInto(JobActivityPublisherRecord.class);
+            if(records.size() != 0) {
+                System.out.println(records.get(0).getRecordType());
```
Comment (tbak) [inline titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/store/JooqJobActivityStore.java:109]: logger?

File: titus-supplementary-component/job-activity-history/src/main/resources/db/migration/V1__initial_setup.sql [hunk 6]
```diff
@@ -16,22 +16,30 @@
...
 CREATE TABLE IF NOT EXISTS jobactivity.jobs
 (
-  job_id          VARCHAR (64) NOT NULL,
-  create_time     TIMESTAMP NOT NULL,
-  record_time     TIMESTAMP NOT NULL,
-  job_record_json jsonb,
-  PRIMARY KEY (job_id)
+    job_id          VARCHAR(64) NOT NULL,
+    record_time     TIMESTAMP   NOT NULL,
+    job_record_json varchar(64),
```
Comment (tbak) [inline titus-supplementary-component/job-activity-history/src/main/resources/db/migration/V1__initial_setup.sql:23]: Ideally this should be unlimited size or very high (128KB). We might also consider gziping it to save space.

**Review by tbak — APPROVED** [review]

File: titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/store/JooqJobActivityStore.java (hunk 5 above)
Comment (amitaekbote) [reply titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/store/JooqJobActivityStore.java:109]: oops looks like I missed these

**Pull request merged**